launchlib/config_test.go

```diff
@@ -67,67 +75,57 @@ env:
 jvmOpts:
   - jvmOpt1
   - jvmOpt2
-`)
-	expectedConfig := CustomLauncherConfig{
-		ConfigType:    "java",
-		ConfigVersion: 1,
-		Env: map[string]string{
-			"SOME_ENV_VAR":  "/etc/profile",
-			"OTHER_ENV_VAR": "/etc/redhat-release",
+`,
+			want: CustomLauncherConfig{
+				LauncherConfig: LauncherConfig{
+					ConfigType:    "java",
+					ConfigVersion: 1,
+				},
+				Env: map[string]string{
+					"SOME_ENV_VAR":  "/etc/profile",
+					"OTHER_ENV_VAR": "/etc/redhat-release",
+				},
+				JvmOpts: []string{"jvmOpt1", "jvmOpt2"}},
 		},
-		JvmOpts: []string{"jvmOpt1", "jvmOpt2"}}
-
-	config := ParseCustomConfig(data)
-	if !reflect.DeepEqual(config, expectedConfig) {
-		t.Errorf("Expected config %v, found %v", expectedConfig, config)
-	}
-}
-
-func TestParseCustomConfigWithoutEnv(t *testing.T) {
-	var data = []byte(`
+		{
+			name: "custom config without env",
+			data: `
 configType: java
 configVersion: 1
 jvmOpts:
   - jvmOpt1
   - jvmOpt2
-`)
-	expectedConfig := CustomLauncherConfig{
-		ConfigType:    "java",
-		ConfigVersion: 1,
-		JvmOpts:       []string{"jvmOpt1", "jvmOpt2"}}
-
-	config := ParseCustomConfig(data)
-	if !reflect.DeepEqual(config, expectedConfig) {
-		t.Errorf("Expected config %v, found %v", expectedConfig, config)
-	}
-
-	if config.Env != nil {
-		t.Errorf("Expected environment to be nil, but was %v", config.Env)
-	}
-}
-
-func TestParseCustomConfigWithEnvPlaceholder(t *testing.T) {
-	var data = []byte(`
+`,
+			want: CustomLauncherConfig{
+				LauncherConfig: LauncherConfig{
+					ConfigType:    "java",
+					ConfigVersion: 1,
+				},
+				JvmOpts: []string{"jvmOpt1", "jvmOpt2"}},
+		},
+		{
+			name: "custom config with env placeholder",
+			data: `
 configType: java
 configVersion: 1
 env:
   SOME_ENV_VAR: '{{CWD}}/etc/profile'
 jvmOpts:
   - jvmOpt1
   - jvmOpt2
-`)
-
-	expectedConfig := CustomLauncherConfig{
-		ConfigType:    "java",
-		ConfigVersion: 1,
-		Env: map[string]string{
-			"SOME_ENV_VAR": "{{CWD}}/etc/profile",
+`,
+			want: CustomLauncherConfig{
+				LauncherConfig: LauncherConfig{
+					ConfigType:    "java",
+					ConfigVersion: 1,
+				},
+				Env: map[string]string{
+					"SOME_ENV_VAR": "{{CWD}}/etc/profile",
+				},
+				JvmOpts: []string{"jvmOpt1", "jvmOpt2"}},
 		},
```

Comment:
Don't think alignment is actually off here, but will switch to expanded style to make this more clear.